packages/mobile/src/components/publish-playlist-drawer/PublishPlaylistDrawer.tsx

```diff
@@ -0,0 +1,144 @@
+import { useCallback } from 'react'
+
+import {
+  cacheCollectionsActions,
+  publishPlaylistConfirmationModalUISelectors
+} from '@audius/common'
+import { View } from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
+
+import IconRocket from 'app/assets/images/iconRocket.svg'
+import { Text, Button } from 'app/components/core'
+import { useManualToast } from 'app/hooks/useManualToast'
+import { useToast } from 'app/hooks/useToast'
+import { makeStyles } from 'app/styles'
+import { useColor } from 'app/utils/theme'
+
+import { useDrawerState } from '../drawer'
+import Drawer from '../drawer/Drawer'
+import LoadingSpinner from '../loading-spinner/LoadingSpinner'
+
+const { publishPlaylist } = cacheCollectionsActions
+const { getPlaylistId } = publishPlaylistConfirmationModalUISelectors
+
+const messages = {
+  drawerTitle: 'Make Public',
+  drawerBody:
+    'Are you sure you want to make this playlist public? It will be shared to your feed and your followers will be notified.',
+  buttonConfirmText: 'Make Public',
+  buttonCancelText: 'Cancel',
+  publishingPlaylistText: 'Making Public...',
+  publishedPlaylistText: 'Your playlist is now public!'
+}
+
+const useStyles = makeStyles(({ palette, spacing, typography }) => ({
+  title: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    gap: spacing(2),
+    alignItems: 'center',
+    paddingVertical: spacing(6),
+    borderBottomColor: palette.neutralLight8,
+    borderBottomWidth: 1
+  },
+  titleText: {
+    textTransform: 'uppercase'
+  },
+  container: {
+    marginHorizontal: spacing(4)
+  },
+  body: {
+    margin: spacing(4),
+    lineHeight: spacing(6),
+    textAlign: 'center',
+    color: palette.neutral,
+    fontSize: typography.fontSize.large,
```

Comment:
these styles can also be added directly to text which can sometimes be faster, just an fyi: 

```
<Text fontSize='large' weight='medium' />
```